smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -84,28 +84,56 @@ contract MixinPurchase is
     uint inMemoryKeyPrice = keyPrice;
     (discount, tokens) = unlockProtocol.computeAvailableDiscountFor(_recipient, inMemoryKeyPrice);
```

Comment:
Hmm on second thought it seems we should apply the hook discount before calling unlockProtocol